### PR TITLE
filter unassigned data with an additional action

### PR DESCRIPTION
**Pull request by Nil20**

### CODE REVIEW

File: packages/client/src/workqueue/reducer.ts [hunk 1]
```diff
@@ -129,6 +133,78 @@ export function updateRegistrarWorkqueue(
...
+    currentlyDownloadedDeclarations = savedDeclarations
+  else {
+    unassignedDeclarations = workqueueDeclarations
+      .filter(
+        (dec) =>
+          dec &&
+          dec.registration &&
+          (dec.registration.assignment ||
+            !hasStatusChanged(dec, savedDeclarations)) &&
+          declarationIds.includes(dec.id) &&
```
Comment (Zangetsu101) [inline packages/client/src/workqueue/reducer.ts:172]: This should not be needed as we are mapping to them just after this filter

**Review by Zangetsu101 — CHANGES_REQUESTED** [review]

File: packages/client/src/workqueue/reducer.ts [hunk 2]
```diff
@@ -157,6 +233,41 @@ async function getWorkqueueOfCurrentUser(): Promise<string> {
...
+        SUBMISSION_STATUS.REGISTERING,
+        SUBMISSION_STATUS.REJECTING,
+        SUBMISSION_STATUS.ARCHIVING,
+        SUBMISSION_STATUS.CERTIFYING,
+        SUBMISSION_STATUS.ISSUING,
+        SUBMISSION_STATUS.REQUESTING_CORRECTION
+      ].includes(declarationInStore.submissionStatus as SUBMISSION_STATUS)
+    )
+  return true
+}
```
Comment (naftis) [inline packages/client/src/workqueue/reducer.ts:257]: 👍🏻makes the
```
        hasStatusChanged(dec, savedDeclarations) &&
        isNotSubmitting(dec, savedDeclarations)
```
part readable!

File: packages/client/src/workqueue/reducer.ts [hunk 3]
```diff
@@ -129,6 +133,84 @@ export function updateRegistrarWorkqueue(
...
+    .filter((maybeDeclaration): maybeDeclaration is IDeclaration =>
+      Boolean(maybeDeclaration)
+    )
+
+  currentlyDownloadedDeclarations = savedDeclarations.filter(
+    (dec) =>
+      !unassignedDeclarations
+        .map((unassigned) => unassigned.id)
+        .includes(dec.id)
+  )
```
Comment (Zangetsu101) [inline packages/client/src/workqueue/reducer.ts:187]:
```suggestion
  const unassignedDeclarationIds = unassignedDeclarations.map((unassigned) => unassigned.id)
  currentlyDownloadedDeclarations = savedDeclarations.filter(
    (dec) => !unassignedDeclarationIds.includes(dec.id)
  )
```

File: packages/client/src/workqueue/reducer.ts [hunk 4]
```diff
@@ -129,6 +133,84 @@ export function updateRegistrarWorkqueue(
...
+  const workqueueDeclarations = Object.entries(workqueue.data).flatMap(
+    (queryData) => {
+      return queryData[1].results
+    }
+  ) as Array<GQLEventSearchSet | null>
+
+  let [currentlyDownloadedDeclarations, unassignedDeclarations]: [
+    IDeclaration[],
+    IDeclaration[]
+  ] = [[], []]
```
Comment (Zangetsu101) [inline packages/client/src/workqueue/reducer.ts:156]: Can we remove this please? From what I could see, we are only assigning the values once

File: packages/client/src/workqueue/reducer.ts [hunk 5]
```diff
@@ -129,6 +133,84 @@ export function updateRegistrarWorkqueue(
...
+      !unassignedDeclarations
+        .map((unassigned) => unassigned.id)
+        .includes(dec.id)
+  )
+
+  if (unassignedDeclarations.length > 0)
+    return {
+      currentlyDownloadedDeclarations: savedDeclarations,
+      unassignedDeclarations
+    }
```
Comment (Zangetsu101) [inline packages/client/src/workqueue/reducer.ts:193]: Could you explain this part please?

Comment (Nil20) [reply packages/client/src/workqueue/reducer.ts:193]: if there are no unassigned declarations we don't have to update the indexDB, we can just return it. It actually should be, `(unassignedDeclarations.length === 0)`

Comment (Nil20) [reply packages/client/src/workqueue/reducer.ts:193]: like we do for field agents previously

**Review by Zangetsu101 — APPROVED** [review]

**Review by naftis — APPROVED** [review]

**Pull request merged**